torch/_export/passes/lift_constants_pass.py

```diff
@@ -274,17 +275,35 @@ def lift_constants_pass(
                     with unset_fake_temporarily():
                         constant_val = constant_val.data
                 constant_kind = InputKind.CONSTANT_TENSOR
-                constant_fqn = _get_first_fqn(constant_attrs, constant_val)
-                if constant_fqn is not None:
-                    constant_name = constant_fqn.replace(".", "_")
-                else:
-                    constant_name = f"lifted_tensor_{num_tensor_constants}"
+
+                # First check if the tensor has a custom name set via our naming API
+                custom_name = get_tensor_name(constant_val)
+                if custom_name is not None:
+                    constant_name = custom_name
                     constant_fqn = get_constant_fqn(node, constant_name)
+                    # Ensure uniqueness if custom name conflicts
+                    original_name = constant_name
+                    counter = 1
                     while constant_fqn in used_target_names:
-                        num_tensor_constants += 1
+                        constant_name = f"{original_name}_{counter}"
+                        constant_fqn = get_constant_fqn(node, constant_name)
+                        counter += 1
+                    # Mark this name as used to prevent future conflicts
+                    used_target_names.add(constant_fqn)
+                else:
+                    # Fall back to module attribute name if available
+                    constant_fqn = _get_first_fqn(constant_attrs, constant_val)
+                    if constant_fqn is not None:
+                        constant_name = constant_fqn.replace(".", "_")
+                    else:
+                        # Use default naming as last resort
                         constant_name = f"lifted_tensor_{num_tensor_constants}"
                         constant_fqn = get_constant_fqn(node, constant_name)
-                    num_tensor_constants += 1
+                        while constant_fqn in used_target_names:
+                            num_tensor_constants += 1
+                            constant_name = f"lifted_tensor_{num_tensor_constants}"
+                            constant_fqn = get_constant_fqn(node, constant_name)
+                        num_tensor_constants += 1
```

Comment:
Duplicate code with the above?

I'd suggest having a single helper to get the name, rather than both `get_tensor_name` and `get_constant_fqn`.   And remove the `if custom_name is not None` branch from this function.